noise/conn.go

```diff
@@ -351,6 +355,21 @@ func (c *Conn) IsRemoteAuthenticated() bool {
 	return c.isRemoteAuthenticated
 }
 
+// StaticKey returns the static key of the remote peer. It is useful in case the
+// static key is only transmitted during the handshake.
+func (c *Conn) StaticKey() ([]byte, error) {
+	if !c.IsRemoteAuthenticated() {
```

Comment:
Are there usecases where you would still want to obtain the public key, even if the peer has not been authenticated yet? Maybe if you're planning to authenticate post-handshake (using a short-authentication string or something like that)

Reply:
True. I was not sure about that, so I just thought going to the safest path. But Is it really possible to get the static key without verifying ? This library panics if there's no `PublicKeyVerifier` in the `Config` IIRC .

Reply:
true.

Another scenario I can think of: setting up the `PublicKeyVerifier()` based on the static public key you received. But this is currently not possible with this library.